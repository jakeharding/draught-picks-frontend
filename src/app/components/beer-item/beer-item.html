<!-- Generated template for the BeerItemComponent component -->
<div class="pointer" (click)="beerTap()">
  <ion-row>
    <ion-col size="9" class="ion-text-left">
      <h1>{{beer.name}}</h1>
    </ion-col>
    <ion-col size="3" class="ion-text-right">
      <div *ngIf="beer.recommended.percent_match">
        <span class="percent">{{ beer.recommended.percent_match}}%</span>
        <span class="ion-text-center">match</span>
      </div>
    </ion-col>
  </ion-row>

  <ion-row>
    <ion-col size="9" class="ion-text-left">
      <p>Your rating:</p>
    </ion-col>
  </ion-row>

  <ion-row>
    <ion-col size="8">
      <rating [rating]="beer.rating[0]" [size]="'small'"></rating>
    </ion-col>

    <ion-col size="2" class="ion-text-right">
      <strong *ngIf="beer.abv">{{ beer.abv }}</strong>
      <strong *ngIf="!beer.abv">N/A</strong>
      <p><strong>ABV</strong></p>
    </ion-col>

    <ion-col size="2" class="ion-text-right">
      <strong *ngIf="beer.ibu">{{ beer.ibu }}</strong>
      <strong *ngIf="!beer.ibu">N/A</strong>
      <p><strong>IBU</strong></p>
    </ion-col>
  </ion-row>
  <hr>
</div>
