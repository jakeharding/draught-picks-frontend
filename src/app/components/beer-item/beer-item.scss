:host {
  .percent {
    font-size: 1.5em;
    font-weight: 500;
    display: block;
  }

  hr {
    height: 1px!important;
    background-color: var(--ion-color-medium-tint);
  }

  ion-col {
    padding: 0px 0px !important;
  }

  ion-row {
    p {
      margin: 0px 0px 4px 4px;
    }
  }
}
