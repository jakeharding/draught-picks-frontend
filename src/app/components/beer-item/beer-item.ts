import { Component, Input } from '@angular/core';

/**
 * Generated class for the BeerItemComponent component.
 *
 * See https://angular.io/api/core/Component for more info on Angular
 * Components.
 */
@Component({
  selector: 'beer-item',
  templateUrl: 'beer-item.html',
  styleUrls: ['./beer-item.scss']
})
export class BeerItemComponent {

  @Input() beer;

  constructor() {}
}
