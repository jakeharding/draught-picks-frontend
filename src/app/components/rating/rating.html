<!-- Generated template for the RatingComponent component -->
<ion-row>
  <div (click)="updateRating(1)" [ngClass]="{'color-mug': rating?.rating >= 1, 'no-color-mug': rating?.rating < 1, 'large': size == 'large'}"></div>
  <div (click)="updateRating(2)" [ngClass]="{'color-mug': rating?.rating >= 2, 'no-color-mug': rating?.rating < 2, 'large': size == 'large'}"></div>
  <div (click)="updateRating(3)" [ngClass]="{'color-mug': rating?.rating >= 3, 'no-color-mug': rating?.rating < 3, 'large': size == 'large'}"></div>
  <div (click)="updateRating(4)" [ngClass]="{'color-mug': rating?.rating >= 4, 'no-color-mug': rating?.rating < 4, 'large': size == 'large'}"></div>
  <div (click)="updateRating(5)" [ngClass]="{'color-mug': rating?.rating >= 5, 'no-color-mug': rating?.rating < 5, 'large': size == 'large'}"></div>
</ion-row>
