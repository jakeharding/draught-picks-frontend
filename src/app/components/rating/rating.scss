:host {
  display: flex;
  flex-direction: row;

  .color-mug{
    background-image: url("../../../assets/imgs/rating/mug_color.png");
    background-size: cover;
    height: 30px;
    width: 30px;
    margin-right: 5px;
  }

  .no-color-mug {
    background-image: url("../../../assets/imgs/rating/mug.png");
    background-size: cover;
    height: 30px;
    width: 30px;
    margin-right: 5px;
  }

  div {
    flex: 1;
  }

  .large {
    height: 55px;
    width: 55px;
    margin-right: 10px;
  }

  @media (max-width: 576px) {
    .large {
      height: 50px;
      width: 50px;
      margin-right: 0px;
    }
  }
}
