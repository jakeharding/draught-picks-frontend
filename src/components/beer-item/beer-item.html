<!-- Generated template for the BeerItemComponent component -->
<div class="pointer" [navPush]="'beer-detail'" [navParams]="beer">
  <ion-row>
    <ion-col col-9 text-left>
      <h1>{{beer.name}}</h1>
    </ion-col>
    <ion-col text-right col-3>
      <span class="percent">98%</span>
      <span text-center>match</span>
    </ion-col>
  </ion-row>

  <ion-row>
    <ion-col col-9 text-left>
      <p>Your rating:</p>
    </ion-col>
  </ion-row>

  <ion-row>
    <ion-col col-8>
      <rating [rating]="beer.rating[0]" [size]="small"></rating>
    </ion-col>

    <ion-col col-2 text-right>
      <strong *ngIf="beer.abv">{{ beer.abv }}</strong>
      <strong *ngIf="!beer.abv">N/A</strong>
      <p><strong>ABV</strong></p>
    </ion-col>

    <ion-col col-2 text-right>
      <strong *ngIf="beer.ibu">{{ beer.ibu }}</strong>
      <strong *ngIf="!beer.ibu">N/A</strong>
      <p><strong>IBU</strong></p>
    </ion-col>
  </ion-row>
  <hr>
</div>
