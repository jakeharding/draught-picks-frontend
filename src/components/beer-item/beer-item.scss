beer-item {
  .title {
    font-size: 2em;
  }

  hr {
    height: 1px!important;
  }

  ion-col {
    padding: 0px 0px !important;
  }

  ion-row {
    p {
      margin: 0px 0px 4px 4px;
    }
  }
}
