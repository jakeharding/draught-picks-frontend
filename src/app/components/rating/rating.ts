import { Component, EventEmitter, Input, Output } from '@angular/core';
import BeerRating from '../../models/BeerRating';
import { RatingProvider } from '../../services/rating/rating';
import { ToastController } from '@ionic/angular';

/**
 * Generated class for the RatingComponent component.
 *
 * See https://angular.io/api/core/Component for more info on Angular
 * Components.
 */
@Component({
  selector: 'rating',
  templateUrl: 'rating.html',
  styleUrls: ['./rating.scss']
})
export class RatingComponent {

  constructor(public ratingProvider: RatingProvider, public toastController: ToastController) {}

  private static readonly LARGE = 'large';

  @Input() rating: BeerRating = { beer: 'beerId'} as BeerRating;
  @Input() size = 'small';
  @Output() setRating = new EventEmitter<BeerRating>();

  /**
   * updateRating function
   * Parameters: rating
   * updates the rating of a beer
   */
  updateRating(rating: number) {
    if (this.size === RatingComponent.LARGE) {
      const success = async beerRating => {
        this.rating = beerRating; // Update the beerRating returned from REST API
        const toast = await this.toastController.create({
          message: 'Your rating is saved.',
          duration: 3000,
          position: 'top',
          cssClass: 'success-toast'
        });
        await toast.present();
        this.setRating.emit(beerRating); // Update parent component
      };

      if (this.rating && this.rating.uuid) {
        return this.ratingProvider.partialUpdate({rating, uuid: this.rating.uuid} as BeerRating).then(success);
      } else {
        return this.createRating(this.rating.beer, rating).then(success);
      }
    }
  }

  /**
   * createRating function
   * Parameters: rating
   * creates a new rating for a beer if none exist
   */
  createRating(beer: string, rating: number) {
    const newRating = {beer, rating} as BeerRating;
    return this.ratingProvider.create(newRating);
  }
}
