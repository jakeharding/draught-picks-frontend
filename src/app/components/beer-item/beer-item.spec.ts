/**
 * beer-item.spec.ts
 *
 * Created by jake
 * Created on 11/15/18
 *
 * Test the beer-item component.
 */
import { BeerItemComponent } from './beer-item';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { NavController } from '@ionic/angular';
import { RatingComponent } from '../rating/rating';
import { RatingProvider } from '../../services/rating/rating';
import { ToastController } from '@ionic/angular';
import { ActivatedRoute, Router, RouterModule } from '@angular/router';
import { mockRouter } from '../../../../setup-jest';
import { HttpClient } from '@angular/common/http';

describe('BeerItemComponent', () => {
  let component: BeerItemComponent;
  let fixture: ComponentFixture<BeerItemComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      declarations: [BeerItemComponent, RatingComponent],
      imports: [RouterModule],
      providers: [
        { provide: Router, useValue: mockRouter},
        { provide: ActivatedRoute, useValue: {} },
        { provide: HttpClient, useValue: {} },
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(BeerItemComponent);
    component = fixture.debugElement.componentInstance;
  });

  it('should initialize the BeerItemComponent and match snapshot', () => {
    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });
});
