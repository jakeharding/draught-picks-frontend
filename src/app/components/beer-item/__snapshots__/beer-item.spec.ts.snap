// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerItemComponent should initialize the BeerItemComponent and match snapshot 1`] = `
<beer-item
  navCtrl={[Function Object]}
>
  <div
    class="pointer"
  >
    <ion-row>
      <ion-col
        class="ion-text-left"
        size="9"
      >
        <h1>
          
        </h1>
      </ion-col>
      <ion-col
        class="ion-text-right"
        size="3"
      >
        
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col
        class="ion-text-left"
        size="9"
      >
        <p>
          Your rating:
        </p>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col
        size="8"
      >
        <rating>
          <ion-row>
            <div />
            <div />
            <div />
            <div />
            <div />
          </ion-row>
        </rating>
      </ion-col>
      <ion-col
        class="ion-text-right"
        size="2"
      >
        
        
        <p>
          <strong>
            ABV
          </strong>
        </p>
      </ion-col>
      <ion-col
        class="ion-text-right"
        size="2"
      >
        
        
        <p>
          <strong>
            IBU
          </strong>
        </p>
      </ion-col>
    </ion-row>
    <hr />
  </div>
</beer-item>
`;
